lib/surface/compiler.ex

```diff
@@ -1026,6 +1082,23 @@ defmodule Surface.Compiler do
     IOHelper.compile_error(message, template_meta.file, template_meta.line)
   end
 
+  defp raise_complex_generator(meta) do
+    message = """
+    using <#for>..<#else> only supports single generators without filters.
+
+    Example:
+      ```
+      <#for each={i <- [1, 2, 3]}>
+      ...
+      <#else>
+      ...
+      </#for>
+      ```
```

Comment:
Assuming the error will be in the `<#else>` line, we could have:

```suggestion
    using `<#else>` is only supported when the expression in `<#for>` has a single generator and no filters.

    Example:

      <#for each={i <- [1, 2, 3]}>
        ...
      <#else>
        ...
      </#for>
```

However, it's hard to know if the user will prefer to keep the `else` changing the `for` expression or keep the generators/filters getting rid of the `else`. The user might not even know the concept of generators and filters. 

I'm not sure we're able to provide a better message than that.

Reply:
> Assuming the error will be at the <#else> line

It was at the `<#for>` line, but changed it and added the line number to the test.

You write good error messages 👏